x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/process/NativeControllerTests.java

```diff
@@ -146,6 +159,28 @@ public void testGetNativeCodeInfo() throws IOException, TimeoutException {
         assertEquals("a0d6ef8819418c", nativeCodeInfo.get("build_hash"));
     }
 
+    public void testChildProcessTerminatedMessageIsAudited() throws Exception {
+
+        NamedPipeHelper namedPipeHelper = mock(NamedPipeHelper.class);
+        ByteArrayInputStream logStream = new ByteArrayInputStream(CHILD_TERMINATED_MESSAGE.getBytes(StandardCharsets.UTF_8));
+        when(namedPipeHelper.openNamedPipeInputStream(contains("log"), any(Duration.class))).thenReturn(logStream);
+        ByteArrayOutputStream commandStream = new ByteArrayOutputStream();
+        when(namedPipeHelper.openNamedPipeOutputStream(contains("command"), any(Duration.class))).thenReturn(commandStream);
+        ByteArrayInputStream outputStream = new ByteArrayInputStream("[]".getBytes(StandardCharsets.UTF_8));
+        when(namedPipeHelper.openNamedPipeInputStream(contains("output"), any(Duration.class))).thenReturn(outputStream);
+
+        SystemAuditor auditor = mock(SystemAuditor.class);
+        NativeController nativeController = NativeController.makeNativeController(
+            NODE_NAME,
+            TestEnvironment.newEnvironment(settings),
+            namedPipeHelper,
+            mock(NamedXContentRegistry.class),
+            auditor
+        );
+
+        assertBusy(() -> verify(auditor).error(eq("Child process with PID 10210 was terminated by signal 9")));
```

Comment:
I think end users will find this frustrating. We'll get angry questions about how they are supposed to know what PID 10210 was. It's also going in the opposite direction to where we're trying to go with serverless, by exposing the end user to _more_ low level implementation detail rather than less.

I think we need to remember a mapping of PID to process type, and then make these messages higher level. For example, if we said `Anomaly detection process was terminated by signal 9` then that would be more understandable by the end user. If we get a support case and want to tie up the PID with other details we'll need to ask for the logs anyway.

If you don't have time to make these changes then you can leave the PR in its current state and somebody else can pick it up in the future.